shared/actions/devices.js

```diff
@@ -1,138 +1,192 @@
-      },
-    }
+export function removeDevice (deviceID: string, name: string, currentDevice: boolean): RemoveDevice {
+  return {type: Constants.removeDevice, payload: {deviceID, name, currentDevice}}
+}
 
-    loginPaperKeyRpc({
-      incomingCallMap,
-      callback: (error, paperKey) => {
-        if (error) {
-          dispatch({
-            type: Constants.paperKeyLoaded,
-            payload: error,
-            error: true,
-          })
-        }
-      },
-    })
+export function generatePaperKey (): GeneratePaperKey {
+  return {type: Constants.generatePaperKey, payload: undefined}
+}
+
+function * _deviceListSaga (): SagaGenerator<any, any> {
+  yield put(loadingDevices())
+  try {
+    const devices = yield call(deviceDeviceHistoryListRpcPromise)
+    yield put(({
+      type: Constants.showDevices,
+      payload: devices,
+      error: false,
+    }: ShowDevices))
+  } catch (e) {
+    yield put(({
+      type: Constants.showDevices,
+      payload: e,
+      error: true,
+    }: ShowDevices))
   }
 }
 
-export function removeDevice (deviceID: string, name: string, currentDevice: boolean): AsyncAction {
-  return (dispatch, getState) => {
-    if (currentDevice) {
-      // Revoking the current device uses the "deprovision" RPC instead.
-      const username = getState().config.username
+function * _deviceRemoveSaga (removeAction: RemoveDevice): SagaGenerator<any, any> {
+  // Revoking the current device uses the "deprovision" RPC instead.
+  const {currentDevice, name, deviceID} = removeAction.payload
+  if (currentDevice) {
+    try {
+      const username = yield select(state => state.config && state.config.username)
       if (!username) {
-        console.warn('No username in removeDevice')
-        return
+        const error = {errorText: 'No username in removeDevice'}
+        console.warn(error)
+        yield put(({
+          type: Constants.deviceRemoved,
+          payload: error,
+          error: true,
+        }: DeviceRemoved))
       }
-      loginDeprovisionRpc({
-        param: {username, doRevoke: true},
-        callback: error => {
-          if (error) {
-            dispatch({
-              type: Constants.deviceRemoved,
-              payload: error,
-              error: true,
-            })
-          } else {
-            dispatch({
-              type: Constants.deviceRemoved,
-              payload: null,
-              error: false,
-            })
-          }
-
-          if (!error) {
-            dispatch(loadDevices())
-            dispatch(setRevokedSelf(name))
-            dispatch(navigateTo([], loginTab))
-            dispatch(switchTab(loginTab))
-          }
-        },
-      })
-    } else {
-      revokeRevokeDeviceRpc({
+      yield call(loginDeprovisionRpcPromise, {param: {username, doRevoke: true}})
+      yield put(({
+        type: Constants.deviceRemoved,
+        payload: undefined,
+        error: false,
+      }: DeviceRemoved))
+      yield put(loadDevices())
```

Comment:
Good question!  Again, was in the previous code.  I can't think of why it would be needed.